### PR TITLE
Fsm

### PR DESCRIPTION
**Pull request by stianbm**
Made a Rule class to support fsm
Also added the skeleton of agent
Around half of the code lines are in tests

### CODE REVIEW

File: keypad/tests/test_rules.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+"""File contains unittests for Rule class"""
```
Comment (LudvigHz) [inline keypad/tests/test_rules.py]: Nice, bare endre fra `../tests/..` til `../test/..` for at testene skal kjøre.
https://stackoverflow.com/a/6672873

File: keypad/finite_state_machine/rule.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    is a function that returns true for the signals that trigger the rule"""
+
+    _trigger_state = None
+    _trigger_signal = None
+    _new_state = None
+    _action = None
+
+    def __init__(self, trigger_state: str, trigger_signal, new_state: str, action):
+        if not hasattr(STATES, trigger_state) or not hasattr(STATES, new_state):
+            raise Exception("ILLEGAL STATE CHOSEN")
```
Comment (LudvigHz) [inline keypad/finite_state_machine/rule.py]: Change to `ValueError`

File: keypad/tests/test_rules.py (hunk 1 above)
Comment (stianbm) [reply keypad/tests/test_rules.py]: Done! La til test/__init__.py også!

File: keypad/finite_state_machine/rule.py (hunk 2 above)
Comment (stianbm) [reply keypad/finite_state_machine/rule.py]: Done! Endret tilhørende tester også

**Review by LudvigHz — APPROVED** [review]
lgtm

**Pull request merged**